src/assets/20-tech-terms/20-beginner-tech-terms.md

```diff
@@ -0,0 +1,141 @@
+#### 1. Forward-Porting a commit
+`Forward-porting` is the practice of moving changes from an older branch to a newer branch. That way any fix or update done will be in the main development branch. This includes finding appropriate commits, cherry-picking them to the branch created for it, resolving any conflicts and running tests thoroughly on these changes to make sure nothing broke during integration. This way, a version of the codebase is made consistent and stable across different implementations.
+#### 2. Backporting a commit
+`Backporting` is when software has some patches or updates in a recent version and applied to an older version of the same software. It includes: identifying the patches to be done, adapting those patches to be compatible with older versions of the same product, dealing with any undesired side effects surfaced during the last step and finally deploying the backported patch.
+
+## 12. Version Control System (VCS):
+`VCS` is a platform where changes to a codebase are logged using a tool. Sometimes we may get worried that continuously rewriting the same code and one small mistake could destroy the entire project. But there's a superhero in the developer world ready to save our day: `Version Control Systems (VCS)!` This feature allows any number of developers to work on the same project without clashing and records changes in a history with who did what, and why. Key features of `VCS` include:
+
+1. Version Control : Handles the history of the changes and keeps versioned.
+
+2. Branching and merging - Allows developers to work separately on features or bug fixes.
+
+3. Collaboration: Leads to seamless integration of modifications across multiple developers working on the same codebase.
+
+4. Backup & Restore : If failures and errors occur, we can rollback to the previous version.
+
+5. Solving Issues: In case of overlapping changes, we are able to resolve discrepancies.
+
+`Git`, `Subversion (SVN)`, and `Mercurial` are some of the popular `VCS` tools. For instance, `Git` and platforms like `Github` or `GitLab` provide an opportunity to clone repositories, create branches and merge changes which helps to coordinate to work optimally in terms of teamwork as well as project management.
+
+## 13. Unit Testing:
+`Unit testing` is a testing method in which every component of software is tested to check if it is working accurately in those smaller components. It means catching bugs and other types of errors closer to the time they are introduced which results in having more reliable code and easier maintenance.
+Key Points:
+- `Isolation`: Tests are written for small units of code like functions/methods, separately from other parts of the application.
+- `Automation`: With unit testing, tests are automated and can be executed continuously as often as required to fix issues immediately.
+- `Refactoring Safety`: Verify that changes made don't break the previous functionality of the application.
+
+## 14. SSO:
```

Comment:
with most of the terms you write the full term plus abbreviation in the title...